tutorials/ab3TracingCode.md

```diff
@@ -169,27 +223,60 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
     ...
     ```
 
+1. `EditCommandParser` is created first, annotated with [label 3](#3-addressbookparser-editcommandparser)
+
+1. `AddressBookParser` now calls `parse(...)` of the newly created `EditCommandParser`, highlighted in the diagram as well, passing control to
+   `EditCommandParser`, annotated with [label 4](#3-addressbookparser-editcommandparser)
+
+
+### 4. EditCommandParser -> EditCommand
+
+<annotate src="images/tracing/LogicSequenceDiagramWithMainWindow.png" width="900" alt="Sample Image">
+  <!-- Set Legend to both -->
+<a-point x="82%" y="51%"  label="5"/>
+<a-point x="50%" y="64%"  color = "RED" label="6"/>
+<a-point x="26%" y="71%"  color = "RED" label="7"/>
+</annotate>
+
+
 1. Let’s see what `EditCommandParser#parse()` does by stepping into it. You might have to click the 'step into' button multiple times here because there are two method calls in that statement: `EditCommandParser()` and `parse()`.
 
    <box type="tip" seamless>
 
    **Intellij Tip:** Sometimes, you might end up stepping into functions that are not of interest. Simply use the `step out` button to get out of them!
    </box>
 
-1. Stepping through the method shows that it calls `ArgumentTokenizer#tokenize()` and `ParserUtil#parseIndex()` to obtain the arguments and index required.
+1. Stepping through the method, `EditCommandParser#parse()`, shows that it calls `ArgumentTokenizer#tokenize()` and `ParserUtil#parseIndex()` to obtain the arguments and index required.
+
 
 1. The rest of the method seems to exhaustively check for the existence of each possible parameter of the `edit` command and store any possible changes in an `EditPersonDescriptor`. Recall that we can verify the contents of `editPersonDesciptor` through the 'Variables' window.<br>
-   ![EditCommand](images/tracing/EditCommand.png)
+   ![EditCommand](images/tracing/EditCommandAnnotated.png)
+
+1. Note the index and the contents of `EditPersonDescriptor`.
+   In this case, since the argument contains only name, only the `Name` field of
+   `EditPersonDescriptor` is set.
```

Comment:
A suggestion to remove (redundant) whitespace.

```suggestion
1. Note the index and the contents of `EditPersonDescriptor`. In this case, since the argument contains only name, only the `Name` field of `EditPersonDescriptor` is set.
```

Reply:
Thanks